1.2.5

miniwolf opened the pull request:
- Introduce OnGeneratedCSProjectFiles, OnGeneratedCSProject and OnGeneratedSlnSolution callbacks.
- Always use forward slash in source paths
- Analyzers use absolute paths
- Ruleset files for roslyn analyzers
- Extra snap search paths on Ubuntu
- Specific c# language version for specific unity versions
- No longer hide .gitignore in VSCode file explorer

goncalo committed:
Fix paths for source files during project generation

andrews-unity committed:
Replicating PR and adding tests

andrews-unity committed:
Fixing the Callbacks messing with other tests kind of hacky but works

demelev committed:
ProjectGeneration: Fixes file path to analyzers installed from UPM.

demelev committed:
ProjectGeneration: Use ruleset file for csproj files.

demelev committed:
ProjectGeneration: Fixes line ending for CodeAnalysisRuleset.

demelev committed:
Uses Relative path for Ruleset file, fixes local packages csproj generation.

sandy-carter-unity committed:
Add extra snap search path

On Ubuntu Software for 20.04, installing code puts it in `/snap/bin/`.
Without this extra path, unity fails to show VSCode as External Script Editor

ElmarJ committed:
Set correct C# language version for Unity version.
Fixes #12

ElmarJ committed:
Modify tests to match correct C#-version

miniwolf committed:
Change CI responder

hmans committed:
No longer hide .gitignore from VS Code's explorer

unity-cla-assistant commented:
[![CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/pull/badge/not_signed)](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.ide.vscode?pullRequest=18) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.ide.vscode?pullRequest=18) before we can accept your contribution.<br/>**4** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: demelev<br/>:white_check_mark: hmans<br/>:white_check_mark: ElmarJ<br/>:white_check_mark: goncalo<br/>:x: andrews-unity<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.ide.vscode?pullRequest=18) it.</sub>

miniwolf committed:
Fix up project generation

language version retrieved from Unity rather than hardcoded
Ensure absolute paths in analyzers

miniwolf committed:
Validation suite gives a false negative on API change on 2019.2
